### PR TITLE
Update support platform doc for Apple M1 issue

### PR DESCRIPTION
**Pull request by TingluoHuang**
Update doc base on https://github.com/actions/runner/issues/1055

### CODE REVIEW

File: docs/start/envosx.md [hunk 1]
```diff
@@ -5,6 +5,12 @@
 ## Supported Versions
 
   - macOS High Sierra (10.13) and later versions
+ 
+## Apple Silicon M1
 
+The runner currently is not supported on device with Apple M1 chip.
```
Comment (hross) [inline docs/start/envosx.md:11]:
```suggestion
The runner is currently not supported on devices with an Apple M1 chip.  
```

File: docs/start/envosx.md [hunk 2]
```diff
@@ -5,6 +5,12 @@
 ## Supported Versions
 
   - macOS High Sierra (10.13) and later versions
+ 
+## Apple Silicon M1
 
+The runner currently is not supported on device with Apple M1 chip.  
+We are waiting for .NET to supports it.
```
Comment (hross) [inline docs/start/envosx.md:12]:
```suggestion
We are waiting for official .NET support. You can read more here about the [current state of support here](https://github.com/orgs/dotnet/projects/18#card-56812463).
```

**Review by hross — CHANGES_REQUESTED** [review]
I added some wording/formatting suggestions

**Comment by TingluoHuang** [comment]
@hross Updated

**Review by hross — APPROVED** [review]

**Pull request merged**